go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/ONSdigital/dp-dataset-api
 
-go 1.20
+go 1.21
+
+toolchain go1.21.5
```

Comment:
Just questioning as this mismatches the CI go version (1.21.7) in `ci/build.yml` etc. I am still not fully familiar with the new `toolchain` keyword (other than a high-level understanding of its purpose) so I don't know if this will cause any issue. I understand this is as it is because you have 1.21.5 locally but I'm not sure what effect it will have on the built artefacts. Probably nothing needs fixing but I'm just raising it in case it has any significance.

Reply:
I have now amended the versions to be 1.21.5 in the dockerfiles

Reply:
I don't think this is the right approach. We should always try to use the most recent (and hence most secure) versions wherever possible. So downgrading the ci versions just to overcome thins doesn't sit right with me. If you update your local go (eg. `brew upgrade go`) then re-tidy the mod does it work better?